### ISSUE TITLE
Split collection in two by predicate

### ISSUE THREAD

**Issue by Alexander882** [morelinq/MoreLINQ#264]
Sometimes, you need to handle the elements of a collection differently, based on whether they satisfy a predicate or not.
To do this, you could just perform a `Where` operation on the source collection to get the elements that satisfy the predicate, and another `Where` to get the elements that don't satisfy the predicate.
This, however, requires multiple enumerations of the source collection, and writing the predicate twice, with and without a `not` operator.
I'd propose a new operator `SplitBy` that does this.

```csharp
public static PredicatedCollections<T> SplitBy<T>(this IEnumerable<T> source, Predicate<T> predicate)
```
Returning a strongly-typed object with the two collections:

```csharp
public class PredicatedCollections<T>
{
    public IEnumerable<T> SatisfiedCollection { get; }
    public IEnumerable<T> UnsatisfiedCollection { get; }
}
```

The result could be calculated eagerly (returning `IList`s) or lazily.

### Example
```csharp
var numbers = new [] { -10, 5, -1, 3 };
var result = numbers.SplitBy(n => n >= 0);
var positives = result.SatisfiedCollection; // [5, 3]
var negatives = result.UnsatisfiedCollection; // [-10, -1]
```

**Comment by leandromoh** [comment]
A similar feature is proposed in this [PR](https://github.com/morelinq/MoreLINQ/pull/210)

**Comment by atifaziz** [comment]
The implementation in #210 is quite baked but missing unit tests. The [next release](https://github.com/morelinq/MoreLINQ/milestone/4) is currently slated for end of March, which might be too tight for getting this in there. If someone want to step in and help write the unit tests (much appreciated) then I can assign this to the [release after next](https://github.com/morelinq/MoreLINQ/milestone/5). Otherwise, I'll try to get round to it myself in April.

**Comment by atifaziz** [comment]
Closed by 89dcf084c51ad41217b89cbdcc2dc4f67b7f36ca

**Comment by atifaziz** [comment]
Available for testing in [2.4.0 beta 2](https://www.nuget.org/packages/morelinq/2.4.0-beta-02).

[Documentation also available online](https://morelinq.github.io/2.4/ref/api/html/Overload_MoreLinq_MoreEnumerable_Partition.htm)
